ngraph/core/shape_inference/include/assign_shape_inference.hpp

```diff
@@ -0,0 +1,31 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#pragma once
+#include <openvino/op/assign.hpp>
+#include "shape_infer_utils.hpp"
+namespace ov {
+namespace op {
+namespace v3 {
+
+template <class T>
+void shape_infer(const Assign* op, const std::vector<T>& input_shapes, std::vector<T>& output_shapes) {
+    NODE_VALIDATION_CHECK(op, input_shapes.size() == 1 && output_shapes.size() == 1);
+    const auto& input_shape = input_shapes[0];
+    if (input_shape.is_static())
```

Comment:
Those variable info check are skipped because I suppose those info is checked outside the `shape_infer`. But you mention that the CPU side may not have such check, then it makes sense that we include those check inside `shape_infer`